### ISSUE TITLE
Wysiwyg does not work with MW1.26.xx (fix available in branch wysiwyg_mw126_rsl)

### ISSUE THREAD

**Issue by riparap** [Mediawiki-wysiwyg/WYSIWYG-CKeditor#81]
Could it be that the main reason to problems of wysiwyg with MW 1.26.x is this: 
BREAKING CHANGE: mediawiki.legacy.ajax has been removed, following a lengthy deprecation period ([see details here](https://www.mediawiki.org/wiki/Release_notes/1.26)). There seems to be ajax related calls in programs of wysiwyg. If this is the problem, does anybody have any idea how this could be fixed? .... [this page](https://www.mediawiki.org/wiki/ResourceLoader/Legacy_JavaScript) has some related information. Or am I on wrong tracks here?

**Comment by riparap** [comment]
There are two problems:
A. Calls like 'sajax_do_call' will not work anymore. They have to be replaced, $.post( mw.util.wikiScript(), {  action: 'ajax', .... seems to work, but I have problems with returned format of data. It seems to be plain text, because result.responseText will not work. Is there a way to define format in call?

B. Problem when wiki page is reopened in wysiwyg, see #82

**Comment by ricordisamoa** [comment]
From an outside perspective, this extension's tech debt seems to have piled up in a very dramatic way. Just my 2 cents.

**Comment by riparap** [comment]
Could you point out what these tech debts are and if possible the preferable way to fix them ?

It will most likely be so that I do not have enough expertese to fix them all but to get a list of them would be at least a good start point.

**Comment by ricordisamoa** [comment]
You need someone who is experienced with both MW and this extension.

**Comment by riparap** [comment]
Seems that there are not so many that kind of persons available or then they are just silent or do not want to share they knowledge :) .... I had to try to study this myself and seems like I was able to fix this. I am not sure if I did everything by the book because once again this was done more by trial and error than clear vision about the solution... few beers also helped me at most desperate times :). Pile of  tech debt  with this extension is not so huge anymore I hope. 

There are still some style information injected into page directly without resourceloader because I could not figure out how resourceloader could be used, because styles seems to be somehow user specific. 

Branch [wysiwyg_mw126_rsl](https://github.com/Mediawiki-wysiwyg/WYSIWYG-CKeditor/tree/wysiwyg_mw126_rsl) contains latest modification for wysiwyg using MW1.26 and resourceloader (Version 1.5.6_0 [B551+26.03.2016]).

**Comment by riparap** [comment]
Source- mode of CKeditor works (branch wysiwyg_mw126_rsl, Version 1.5.6_0 [B551+17.04.2016]), but so far I have been unable to make it work using resourceloader. Workaround requires usage of variables
$wgWYSIWYGSourceMode = true and $wgResourceLoaderDebug = true in LocalSettings.php.

**Comment by ltuo** [comment]
Installed this branch on MediaWiki 1.26.2 and enabled it in LocalSettings.php. MediaWiki failed to start and there is a message in httpd/error.log as below:
`[Fri Jun 03 15:25:34 2016] [error] [client a.b.c.d] PHP Parse error:  syntax error, unexpected '[' in /var/www/html/mediawiki-1.26.2/extensions/WYSIWYG/CKeditor.body.php on line 628.`

`Line 628:                                 'fck_mv_plg_strtr_span' => [],         //16.01.15 RL`

Added '' to enclose [] for workaround temporarily:
`'fck_mv_plg_strtr_span' => '[]',         //16.01.15 RL`

It works but I don't know if this is the right fix.

**Comment by riparap** [comment]
Thanks for posting the solution. I think single quotes is the right way to fix it.

**Comment by ricordisamoa** [comment]
`[]` (short array syntax) is the same as `array()` but only since PHP 5.4.

**Comment by riparap** [comment]
Related php code  is creating javascript varibles using MW function

``` php
 $wgOut->addJsConfigVars ( array(
....
'fck_mv_plg_strtr_span' => '[]',
....
)
```

Variable `fck_mv_plg_strtr_span` should be array in javascript... so how should this be fixed?

**Comment by ricordisamoa** [comment]
It seems that `fck_mv_plg_strtr_span` is already set to empty array in a JavaScript file... otherwise the old array syntax should work with PHP 5.3.

``` php
$wgOut->addJsConfigVars ( array(
....
'fck_mv_plg_strtr_span' => array(),
....
) );
```

**Comment by riparap** [comment]
Thanks, I have included this in Version 1.5.6_0 [B551+04.06.2016].
